apps/console/src/features/roles/components/application-roles.tsx

```diff
@@ -132,6 +133,31 @@ export const ApplicationRoles: FunctionComponent<ApplicationRolesSettingsInterfa
         }
     }, [ roleAudience ]);
 
+    /**
+     * Send a request to update roles when one of the role audience radio buttons is selected. 
+     */
+    useEffect(() => {
+
+        /**
+         * This if block prevents the roles being updated unless shouldUpdateRoleAudience is specifically set to true.
+         * It prevents the roles from being updated in unwanted scenarios such as in the initial render of the
+         * component.
+         */
```

Comment:
Comments explaining the code should be unnecessary as the code itself should be self-explanatory. Please check and fix the other places as well.

Ref: https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/